### PR TITLE
fix: [CDS-52847]: PAT support in Jira connector

### PR DESCRIPTION
**Pull request by naman-goenka**
## Describe your changes
- supported PAT auth for Jira connector behind FF
- Only if FF is on, we will send newer beans (auth) to delegate for connectivity test and heartbeat for backward compatibility for old delegates.
- Testing : FF Check for creating PAT connector, Jira sanity when FF on and all services deployed, when FF is off and older delegate should support old connector, when FF is on newer delegate should support old connector

Newer Yaml
```
connector:
  name: JiraServerPat
  identifier: JiraServerPat
  description: ""
  type: Jira
  spec:
    jiraUrl: https://jira.dev.harness.io/
    auth:
      type: PersonalAccessToken
      spec:
        patRef: account.pat
```

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/44364)

### CODE REVIEW

**Comment by naman-goenka** [comment]
trigger messagecheck

**Comment by naman-goenka** [comment]
trigger codeformat

**Review by nishantsaini20 — COMMENTED** [review]
> Reviewed 1 of 1 files at r3.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44364)* status: 34 file(s) to review. 0 of 2 approvals obtained.

**Comment by naman-goenka** [comment]
trigger codeformat

File: 970-api-services-beans/src/main/java/io/harness/jira/JiraInternalConfig.java [hunk 1]
```diff
@@ -15,15 +15,21 @@
...
 public class JiraInternalConfig {
   String jiraUrl;
-  String username;
-  @ToString.Exclude String password;
+  @Deprecated String username;
+  @ToString.Exclude @Deprecated String password;
+  String authToken;
 
   public String getJiraUrl() {
+    //    JiraInternalConfig.builder().authToken("d");
```
Comment (naman-goenka) [inline 970-api-services-beans/src/main/java/io/harness/jira/JiraInternalConfig.java:32]: delete this comment

**Review by puthrayaharness — APPROVED** [review]

**Review by puthrayaharness — DISMISSED** [review]
> Reviewed 20 of 27 files at r1, 2 of 8 files at r2, 1 of 3 files at r4, 11 of 11 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44364)* status: 34 file(s) to review. 1 of 2 approvals obtained.

**Comment by naman-goenka** [comment]
trigger ut3

**Review by trunapushpa — DISMISSED** [review]

**Review by trunapushpa — DISMISSED** [review]

**Review by sribalij — APPROVED** [review]
> Reviewed 20 of 27 files at r1, 2 of 8 files at r2, 1 of 3 files at r4, 10 of 11 files at r5, 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44364)* status: 1 file(s) to review. 1 of 2 approvals obtained.

**Review by vtxorxwitty — APPROVED** [review]
> Reviewed 20 of 27 files at r1, 2 of 8 files at r2, 1 of 3 files at r4, 10 of 11 files at r5, 1 of 1 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44364)* status: 1 file(s) to review. 1 of 2 approvals obtained.

**Review by shivamnegi94 — APPROVED** [review]

**Comment by naman-goenka** [comment]
trigger ut2

**Pull request merged**